tests/test_builds.py

```diff
@@ -66,18 +56,82 @@
 }
 """
 
+params = [
+    # args, kwargs, mock_body, expectations
+    (
+        (),  # no args , no kwargs, expect to get everything with default fields
+        {},
+        # this is not automatically done, need to manually input the expected result
+        filter_builds_response(fields=['number', 'url'], start=1, end=5),
+        {
+            'length': 5,
+            'idx': 4,
+            'url': 'https://localhost:8080/job/dev/job/testgene/2/',
+            'number': 2,
+            'request_url': 'http://server//job/job/api/json?tree=allBuilds%5Bnumber,url%5D',
+        },
+    ),
+    (
+        (['number']),
+        # test a single item without kwargs we expect to fail
+        # because we need to use kwargs for fields
+        {},
+        filter_builds_response(fields=['number'], start=1, end=5),
+        {
+            'length': 5,
+            'idx': 4,
+            'number': 2,
+            'request_url': 'http://server//job/job/api/json?tree=allBuilds%5Bnumber%5D',
+            'error': TypeError
+        }
+    ),
+    (
+        (),  # test selections with start only
+        {'fields': ['number', 'url', 'timestamp'], 'start': 3},
+        filter_builds_response(fields=['number', 'url', 'timestamp'], start=3, end=5),
+        {
+            'length': 3,
+            'idx': 1,
+            'number': 3,
+            'request_url':
+                'http://server//job/job/api/json?tree=allBuilds%5Bnumber,url,timestamp%5D%7B3,%7D'
+        }
+    ),
+    (
+        (),  # test selections with end only
+        {'fields': ['number', 'url', 'timestamp'], 'end': 2},
+        filter_builds_response(fields=['number', 'url', 'timestamp'], start=4, end=5),
+        {
+            'length': 2,
+            'idx': 2,
+            'number': 1,
+            'request_url':
+                'http://server//job/job/api/json?tree=allBuilds%5Bnumber,url,timestamp%5D%7B,2%7D'
+        }
+    )
+]
+
 
+@pytest.mark.parametrize('args,kwargs,mock_body,expectations', params)
 @responses.activate
-def test_get(client):
+def test_get(client, args, kwargs, mock_body, expectations):
```

Comment:
Thanks for such serious tests, but I think it over complicated.

It's enough just to check few simple tests cases or correctly formed **relative** URL.

Pinning on `http://server...` in test is incorrect due to it's additional dependency, for example if I will check if conftest.py  `http://server` to something else, your tests will be failed, so it's more correctly just to check quiery parameters instead of full absolute URL.